[40기 김윤아]Add: login 기능 구현

longyuna opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [ ] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 로그인 페이지 api 연결 

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
- 이메일에 '@'가 있고, 비밀번호가 8자 이상이면 로그인 후 main페이지로 이동



<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- fetch문의 복습
- get, post method의 차이를 알게되었다.

<br />

## :: 기타 질문 및 특이 사항
- ferct문의 돌아가는 방법을 좀 더 알고싶습니다.
- 회원가입 되지 않은 회원에게는 '존재하지 않는 회원입니다'하는 alert 창을 띄우고 싶습니다.
- 정규식을 하고 싶어 구글링 해보았지만, 원하는 값이 나오지 않습니다.

longyuna committed:
Add: login 기능 구현

longyuna committed:
modify: handleLogin 함수에 fetch 적용

longyuna committed:
Modify: else if 조건문 변경

sodalite1204 commented on src/pages/Login/Login.js:
```diff
@@ -1,23 +1,71 @@
+            value={pw}
+          />
+          <button
+            onClick={e => {
+              e.preventDefault();
+              fetch('http://10.58.52.125:8000/users/signin', {
+                method: 'POST',
+                headers: { 'content-Type': 'application/json;charset=utf-8' },
+                body: JSON.stringify({
+                  email: id,
+                  password: pw,
+                }),
+              })
+                .then(res => res.json())
+                .then(data => {
+                  if (data.access_token) {
+                    localStorage.setItem('token', data.access_token);
+
+                    alert('환영합니다!');
+                    navigate('/main');
+                  } else if (data.message === 'specified user does not exist') {
+                    alert('존재하지 않는 유저입니다.');
+                  }
+                });
+            }}
```
가독성이 너무 떨어지네요.  함수는 따로 분리해주세요.

sodalite1204 commented on src/pages/Login/Login.js:
```diff
@@ -1,23 +1,71 @@
+  const navigate = useNavigate();
+
+  const goMainOk = id.includes('@') && pw.length >= 8;
+  const goMainBtn = goMainOk ? false : true;
```
필요없는 코드 같습니다.

sodalite1204 commented on src/pages/Login/Login.js:
```diff
@@ -1,23 +1,71 @@
+
+  const navigate = useNavigate();
+
+  const goMainOk = id.includes('@') && pw.length >= 8;
```
goMainOk => isLoginValidatation 정도로 바꾸는 게 어떨까요?

sodalite1204 commented on src/pages/Login/Login.js:
```diff
@@ -1,23 +1,71 @@
+                });
+            }}
+            className={goMainOk ? 'login-button' : 'login-button-no'}
+            disabled={goMainBtn}
```
```suggestion
            disabled={!goMainOk}
```
not연산자 사용하시면 되는데 일단, goMainOk 변수명이 어썸하지 않아서 변경부터 해주세요.

sodalite1204 reviewed (commented):
리뷰 확인해주시고, 정규식 표현 반영해주시면 좋을 듯 합니다!

longyuna committed:
Modify: login 기능 구현 수정
- fetch 함수 분리
- 변수명 변경 (goMainOk => isLoginValidatation)
- disabled not 연산자 사용
- data.message 변경

sodalite1204 commented:
정규 표현식 적용도 꼭 해보세요!
